### PR TITLE
Close quarters mimery

### PR DESCRIPTION
**Pull request by heepox**
## About The Pull Request
My first PR please be gentle and give me some feedback

Mime traitor martial art...
Might need to tweak some values and such no sure if its too powerful or too weak.

SURPRISE_SPIN_COMBO "DH" Makes the dude do the spin emote and makes them confused for a time (still gotta tweak confusion time i think)

TOUNGE_PULL_COMBO "GHG" Tounge damage and silence for a bit (cannot be performed if the target does not have a tounge)

THROAT_PUNCH_COMBO "HHG" like krav maga lung punch but uhh weaker and without the CLAMP causing no breath just regular breathloss instead.

ARM_PULL_COMBO "GD" like carp wrist wrench but weaker stun and no shoving stunbaton man away (does not fit maybe delete)

MIME_SPECIAL_COMBO "DGGHHD" Rip out tounge destroying it and lots of bleeding and damage and a stun so its a 6 long combo (not sure about qdeling tounge) (cannot be performed if the target does not have a tounge)

Currently 9TC just so you can have it and advanced mimery.
## Why It's Good For The Game

Just one more mime traitor item making the mime have a stunning TWO unique traitor items

## Changelog
:cl:
add: Close quarters Mimery so mimes have something more than advanced mimery and potato
/:cl:

### CODE REVIEW

**Comment by Archanial** [comment]
Martial arts are awful and combos are bad, we shouldn't have more of them to be honest.

File: code/datums/martial/close_quarters_mimery.dm [hunk 1]
```diff
@@ -0,0 +1,136 @@
+#define SURPRISE_SPIN_COMBO "DD" //
+#define TOUNGE_PULL_COMBO "GHG" //
+#define THROAT_PUNCH_COMBO "HHG" //
+#define ARM_PULL_COMBO "GD" //
+#define MIME_SPECIAL_COMBO "DGHGHD" //
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]:
```suggestion
#define SURPRISE_SPIN_COMBO "DD"
#define TOUNGE_PULL_COMBO "GHG"
#define THROAT_PUNCH_COMBO "HHG"
#define ARM_PULL_COMBO "GD"
#define MIME_SPECIAL_COMBO "DGHGHD"
```

File: code/datums/martial/close_quarters_mimery.dm [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	name = "Close Quarters Mimery"
+	id = MARTIALART_CQM
+	allow_temp_override = FALSE
+	help_verb = /mob/living/carbon/human/proc/cqm_help
+
+/datum/martial_art/cqm/proc/check_streak(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(findtext(streak,SURPRISE_SPIN_COMBO))
+		streak = ""
+		headTwist(A,D)
+		return 1
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]:
```suggestion
		return TRUE
```

File: code/datums/martial/close_quarters_mimery.dm [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	if(findtext(streak,TOUNGE_PULL_COMBO))
+		streak = ""
+		toungePull(A,D)
+		return 1
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]:
```suggestion
		return TRUE
```

File: code/datums/martial/close_quarters_mimery.dm [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+	if(findtext(streak,THROAT_PUNCH_COMBO))
+		streak = ""
+		throatPunch(A,D)
+		return 1
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]:
```suggestion
		return TRUE
```

File: code/datums/martial/close_quarters_mimery.dm [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+	if(findtext(streak,ARM_PULL_COMBO))
+		streak = ""
+		armPull(A,D)
+		return 1
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]:
```suggestion
		return TRUE
```

File: code/datums/martial/close_quarters_mimery.dm [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+	if(findtext(streak,MIME_SPECIAL_COMBO))
+		streak = ""
+		mimeSpecial(A,D)
+		return 1
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]:
```suggestion
		return TRUE
```

File: code/datums/martial/close_quarters_mimery.dm [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+	return 0
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]:
```suggestion
	return FALSE
```

File: code/datums/martial/close_quarters_mimery.dm [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+/datum/martial_art/cqm/proc/surpriseSpin(mob/living/carbon/human/A, mob/living/carbon/human/D)
+if(!D.stat && !D.IsStun() && !D.IsParalyzed())
+	log_combat(A, D, "Surprise spun (Close Quarters Mimery)")
+	D.visible_message("<span class='warning'>[A] spins [D] right round like a record!</span>", \
+		"<span class='userdanger'>[A] spins you right round like a record!</span>")
+	D.emote("spin")
+	D.confused += 20
+	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+	playsound(get_turf(D), 'sound/weapons/thudswoosh.ogg', 30, 1, -1)
+	return 1
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]:
```suggestion
	if(!D.stat && !D.IsStun() && !D.IsParalyzed())
		log_combat(A, D, "Surprise spun (Close Quarters Mimery)")
		D.visible_message("<span class='warning'>[A] spins [D] right round like a record!</span>", \
		"<span class='userdanger'>[A] spins you right round like a record!</span>")
		D.emote("spin")
		D.confused += 20
		A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
		playsound(get_turf(D), 'sound/weapons/thudswoosh.ogg', 30, 1, -1)
		return TRUE
```
It doesn't compile.
You didn't even test it.

File: code/datums/martial/close_quarters_mimery.dm [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+	D.visible_message("<span class='warning'>[A] pulls [D]'s tounge painfully!</span>", \
+		"<span class='userdanger'>[A] pulls your tounge painfully restricting your speech!</span>")
+	D.apply_damage(10, A.dna.species.attack_type, BODY_ZONE_HEAD, def_check)
+	D.adjustOrganLoss(ORGAN_SLOT_TOUNGE, 15, 150)
+	D.Jitter(20)
+	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+	playsound(get_turf(D), 'sound/weapons/blobattack.ogg', 30, 1, -1)
+	if(D.silent <= 10)
+        	D.silent = CLAMP(D.silent + 10, 0, 10)
+	return 1
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]:
```suggestion
	return TRUE
```

File: code/datums/martial/close_quarters_mimery.dm [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+	log_combat(A, D, "Tounge Pulled (Close Quarters Mimery)")
+	D.visible_message("<span class='warning'>[A] pulls [D]'s tounge painfully!</span>", \
+		"<span class='userdanger'>[A] pulls your tounge painfully restricting your speech!</span>")
+	D.apply_damage(10, A.dna.species.attack_type, BODY_ZONE_HEAD, def_check)
+	D.adjustOrganLoss(ORGAN_SLOT_TOUNGE, 15, 150)
+	D.Jitter(20)
+	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+	playsound(get_turf(D), 'sound/weapons/blobattack.ogg', 30, 1, -1)
+	if(D.silent <= 10)
+        	D.silent = CLAMP(D.silent + 10, 0, 10)
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]:
```suggestion
	if(D.silent <= 10)
    	D.silent = CLAMP(D.silent + 10, 0, 10)
```

File: code/datums/martial/close_quarters_mimery.dm [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+//Throat Punch, Prevents breating for a moment, deals oxygen damage and restricts speech for a some time.
+/datum/martial_art/cqm/proc/toungePull(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	log_combat(A, D, "Throat punched (Close Quarters Mimery)")
+	D.visible_message("<span class='warning'>[A] Punches [D]'s throat!</span>", \
+		"<span class='userdanger'>[A] punches your throat restricting your speech and breathing!</span>")
+	playsound(get_turf(A), 'sound/effects/hit_punch.ogg', 50, 1, -1)
+	D.losebreath += 3
+	D.adjustOxyLoss(10)
+	if(D.silent <= 15)
+        	D.silent = CLAMP(D.silent + 15, 0, 15)
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]: same as above, use tabs not spaces....

File: code/datums/martial/close_quarters_mimery.dm [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+	return 1
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]:
```suggestion
	return TRUE
```

File: code/datums/martial/close_quarters_mimery.dm [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+	log_combat(A, D, "Arm pulled (Close Quarters Mimery)")
+	D.visible_message("<span class='warning'>[A] grabs [D]'s arm and pulls it sideways painfully!</span>", \
+		"<span class='userdanger'>[A] grabs and pulls your arm painfully to the side!</span>")
+	D.apply_damage(5, BRUTE, pick(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM))
+	D.dropItemToGround(D.get_active_held_item())
+	D.emote("scream")
+	D.Stun(30)
+	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+	playsound(get_turf(A), 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
+	return 1
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]: That won't compile
see above

File: code/datums/martial/close_quarters_mimery.dm [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+	D.emote("scream")
+	D.Stun(60)
+	D.bleed_rate = CLAMP(D.bleed_rate + 30, 0, 30)
+	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+	playsound(get_turf(A), 'sound/weapons/blobattack.ogg', 50, 1, -1)
+	var/obj/item/organ/tounge/tounge = D.getorganslot(ORGAN_SLOT_TOUNGE)
+	if(D.getorgan(/obj/item/organ/tounge))
+		tounge.Remove(D)
+		qdel(tounge)
+	return 1
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]:
```suggestion
	return TRUE
```

File: code/datums/martial/close_quarters_mimery.dm [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+	D.apply_damage(25, BRUTE, BODY_ZONE_HEAD)
+	D.emote("scream")
+	D.Stun(60)
+	D.bleed_rate = CLAMP(D.bleed_rate + 30, 0, 30)
+	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+	playsound(get_turf(A), 'sound/weapons/blobattack.ogg', 50, 1, -1)
+	var/obj/item/organ/tounge/tounge = D.getorganslot(ORGAN_SLOT_TOUNGE)
+	if(D.getorgan(/obj/item/organ/tounge))
+		tounge.Remove(D)
+		qdel(tounge)
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]:
```suggestion
	var/obj/item/organ/tounge/T = D.getorganslot(ORGAN_SLOT_TOUNGE)
	if(T)
		T.Remove(D)
		qdel(T)
```

File: code/datums/martial/close_quarters_mimery.dm [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+	return 1
+
+/datum/martial_art/cqm/harm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	add_to_streak("H",D)
+	if(check_streak(A,D))
+		return 1
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]:
```suggestion
		return TRUE
```

File: code/datums/martial/close_quarters_mimery.dm [hunk 17]
```diff
@@ -0,0 +1,136 @@
...
+	return ..()
+
+/datum/martial_art/cqm/grab_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	add_to_streak("G",D)
+	if(check_streak(A,D))
+		return 1
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]:
```suggestion
		return TRUE
```

File: code/datums/martial/close_quarters_mimery.dm [hunk 18]
```diff
@@ -0,0 +1,136 @@
...
+	return ..()
+
+/datum/martial_art/cqm/disarm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	add_to_streak("D",D)
+	if(check_streak(A,D))
+		return 1
```
Comment (Archanial) [inline code/datums/martial/close_quarters_mimery.dm]:
```suggestion
		return TRUE
```

File: code/game/objects/items/granters.dm [hunk 19]
```diff
@@ -359,6 +359,33 @@
...
+	..()
+	if(!oneuse)
+		return
+	desc = "It's completely blank."
+	name = "empty scroll"
+	icon_state = "blankscroll"
+
+/obj/item/book/granter/martial/close_quarters_mimery/already_known(mob/user)
+	if(usr && usr.mind)
+		if(!usr.mind.miming)
```
Comment (Archanial) [inline code/game/objects/items/granters.dm]:
```suggestion
	if(user.mind && !user.mind.miming)
```

**Review by Archanial — CHANGES_REQUESTED** [review]
Martial arts could use a rewrite anyway

**Comment by Crossedfall** [comment]
Based on the commits, it looks like you might be trying to do all of this via the web editor. I'd strongly advise looking at the following guide on setting up a proper development environment: https://forums.beestation13.com/t/github-building-the-hive/1334

I'd also recommend getting VSCode to properly edit and test the files.

**Comment by heepox** [comment]
> Martial arts are awful and combos are bad, we shouldn't have more of them to be honest.

if this does not pan out ill propably make an equivalent item to the clowns tae-clown-do shoes for the mime.

**Comment by heepox** [comment]
so i fixed some spelling errors and removed a few tabs and spaces. maybe a few more adjustments and this might be a moment where someone said to me WYCI and i did.

**Comment by heepox** [comment]
currently just testing it out... might need to adjust combos so they dont conflict

**Comment by heepox** [comment]
no more combo conflicts, but mime special is a really nasty attack. gonna lessen the bloodloss unless its fine as a very long combo death dealer. hmm

**Comment by super12pl** [comment]
9 TC for martial art that lets you (practically) permanently silence someone is well.... questionable

File: code/datums/martial/cqm.dm [hunk 20]
```diff
@@ -0,0 +1,140 @@
...
+		log_combat(A, D, "Surprise spun (Close Quarters Mimery)")
+		D.visible_message("<span class='warning'>[A] spins [D] right round like a record!</span>", \
+		"<span class='userdanger'>[A] spins you right round like a record!</span>")
+		D.emote("spin")
+		if(D.confused <= 20)
+			D.confused = CLAMP(D.confused + 10, 0, 20)
+		A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+		playsound(get_turf(D), 'sound/weapons/thudswoosh.ogg', 30, 1, -1)
+		return TRUE
+
```
Comment (Archanial) [inline code/datums/martial/cqm.dm]:
```suggestion
```
random newline

File: code/datums/martial/cqm.dm [hunk 21]
```diff
@@ -0,0 +1,140 @@
...
+//Tounge Pull, Deal 10 brute to the head(reduced by armor(space magic), Deals damage to the targets tounge and restricts speech for a bit.
+/datum/martial_art/cqm/proc/toungePull(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	var/def_check = D.getarmor(BODY_ZONE_HEAD, "melee")
+	log_combat(A, D, "Tounge Pulled (Close Quarters Mimery)")
+	D.visible_message("<span class='warning'>[A] pulls [D]'s tounge painfully!</span>", \
+		"<span class='userdanger'>[A] pulls your tounge painfully restricting your speech!</span>")
+	D.apply_damage(10, A.dna.species.attack_type, BODY_ZONE_HEAD, def_check)
+	D.adjustOrganLoss(ORGAN_SLOT_TONGUE, 15, 200)
+	D.Jitter(20)
+	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
```
Comment (Archanial) [inline code/datums/martial/cqm.dm]: this should have a check to see if target has tongue

File: code/datums/martial/cqm.dm [hunk 22]
```diff
@@ -0,0 +1,140 @@
...
+		D.dropItemToGround(D.get_active_held_item())
+		D.emote("scream")
+		D.Stun(30)
+		return TRUE
+
+	return basic_hit(A,D)
+
+//Mime special, RIP HIS TOUNGE OUT AND CRUSH IT BEFORE HIS EYES! 25 brute to the head and a 6s stun also destroying the targets tounge and making them BLEED!
+/datum/martial_art/cqm/proc/mimeSpecial(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	var/obj/item/organ/tongue/T = D.getorganslot(ORGAN_SLOT_TONGUE)
```
Comment (Archanial) [inline code/datums/martial/cqm.dm:103]: if(T) check should be way above. Whole move doesn't make sense if a person doesn't have tongue in first place.

File: code/game/objects/items/granters.dm [hunk 23]
```diff
@@ -359,6 +359,32 @@
...
+	desc = "It's completely blank."
+	name = "empty scroll"
+	icon_state = "blankscroll"
+
+/obj/item/book/granter/martial/cqm/already_known(mob/user)
+	if(user.mind && !user.mind.miming)
+		to_chat(user, "<span class='warning'>You try to read the scroll but can't comprehend any of it.</span>")
+		return TRUE
+	else
+		return FALSE
```
Comment (Archanial) [inline code/game/objects/items/granters.dm]: This logic is pretty weird. You are returning true if a player does not have mind or is not a mime or else your return false?

**Review by Archanial — CHANGES_REQUESTED** [review]

File: code/game/objects/items/granters.dm (hunk 23 above)
Comment (heepox) [reply code/game/objects/items/granters.dm]: yeah i just need !user.mind.miming so user.mind is not needed. return true means cannot learn in granters false means can learn i know weird.

Comment (heepox) [reply code/game/objects/items/granters.dm]: and yes i tested it... only mimes that have a vow of silence can read it, nobody else

**Comment by heepox** [comment]
> 9 TC for martial art that lets you (practically) permanently silence someone is well.... questionable

losing your tounge does not prevent you from speaking just makes it harder to understand what you are saying

like Help mime bad can turn into h'lp mim' ba'

it basically either replaces a letter or two with aa oo or ' but otherwise leaves the words intact.

also an active radio jammer in your backpack is more effective at silencing people than this martial art and jammer is 3tc

**Comment by Goblinu** [comment]
> 
> 
> > Martial arts are awful and combos are bad, we shouldn't have more of them to be honest.
> 
> if this does not pan out ill propably make an equivalent item to the clowns tae-clown-do shoes for the mime.

If you do make the shoes invisible

**Comment by heepox** [comment]
> > > Martial arts are awful and combos are bad, we shouldn't have more of them to be honest.
> > 
> > 
> > if this does not pan out ill propably make an equivalent item to the clowns tae-clown-do shoes for the mime.
> 
> If you do make the shoes invisible

haha no sprite shoes... eh no i would propably just make them normal shoes or give you a choice list of their appearance to choose from (invisible being one) and then making that the permament sprite for that instance of the item. maybe a mimefu mask instead i dunno.

**Comment by heepox** [comment]
so everything should work as intended now "should"

**Comment by ike709** [comment]
🔑

**Comment by ike709** [comment]
🔑  by fran

**Pull request closed without merging**